rosidl_generator_cpp/resource/msg__traits.hpp.em

```diff
@@ -67,6 +71,32 @@ template<>
 struct has_fixed_size<@(cpp_namespace)@(spec.base_type.type)>
   : std::integral_constant<bool, @(fixed_template_string)>{};
 
+@{
+bounded_template_strings = []
+bounded = True
```

Comment:
That's a fair point. If the initial value is false, the last (non-breaking) condition needs to set it to True. I think that the first 2 conditions still need to set it to False regardless of the initialization of the variable. I can explicitely set it to True in the other blocks if it improves readability. It would result in something like:
```
fixed = False
if len(spec.fields) == 0:
    fixed = True
else:
    for field in spec.fields:
        if field.type.type == 'string':
            fixed = False
            break
        if field.type.is_dynamic_array():
            fixed = False
            break
        if not field.type.is_primitive_type():
            fixed = True
            tmp_fixed_string = "has_fixed_size<{}::msg::{}>::value".format(
                field.type.pkg_name, field.type.type)
            if tmp_fixed_string not in fixed_template_strings:
                fixed_template_strings.append(tmp_fixed_string)
        else:
            fixed = True
```
Is that preferred ?